edgedb/con_utils.py

```diff
@@ -74,7 +74,12 @@
     r'((?:(?:\s|^)-\s*)?\d*\.?\d*)\s*(?i:us(\s|\d|\.|$)|microseconds?(\s|$))',
 )
 INSTANCE_NAME_RE = re.compile(
-    r'^([A-Za-z_]\w*)(?:/([A-Za-z_]\w*))?$',
+    r'^(\w(?:\w|-(?=\w))*)(?:/(\w(?:\w|-(?=\w))*))?$',
```

Comment:
If you're doing `-(?=\w)` you might as well not bother with a look-ahead and instead use the simpler and more explicit approach of actually saying that any `\w` except the first one can be optionally preceded by a single `-`:
`r'^(\w(?:-?\w)*)(?:/(\w(?:-?\w)*))?$'`

It has the same effect of allowing single `-` to be used inside words, but might be a little easier to read for a maintainer (because it uses less obscure regexp features).